components/changelog/ChangelogTrigger.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { graphql } from '@apollo/client/react/hoc';
+import themeGet from '@styled-system/theme-get';
+import styled from 'styled-components';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import { parseToBoolean } from '../../lib/utils';
+
+import Container from '../Container';
+import { Flex } from '../Grid';
+import { withNewsAndUpdates } from '../NewsAndUpdatesProvider';
+import { Dropdown } from '../StyledDropdown';
+import { useUser } from '../UserProvider';
+
+import ChangelogNotificationDropdown from './ChangelogNotificationDropdown';
+
+const CHANGE_LOG_UPDATES_ENABLED = parseToBoolean(process.env.CHANGE_LOG_UPDATES_ENABLED);
+
+const FlameIcon = styled(Flex)`
+  border-radius: 50%;
+  background-color: ${props => themeGet(props.backgroundColor)};
+  background-image: ${props => `url(${props.url})`};
+  background-repeat: no-repeat;
+  background-position: center center;
+  height: 30px;
+  width: 30px;
+  margin-left: 2px;
+`;
+
+const ChangelogTrigger = props => {
+  const { setShowNewsAndUpdates, setChangelogViewDate } = props;
+  const [showChangelogDropdown, setShowChangelogDropdown] = useState(true);
+  const { LoggedInUser, refetchLoggedInUser } = useUser();
+  const hasSeenNewUpdates = LoggedInUser?.hasSeenLatestChangelogEntry;
+
+  const handleShowNewUpdates = async () => {
+    setShowNewsAndUpdates(true);
+    await setChangelogViewDate({ variables: { changelogViewDate: new Date() } });
+    refetchLoggedInUser();
```

Comment:
Manually updating the cache should be safe (it's just one field), and faster than re-fetching the whole thing

See https://www.apollographql.com/docs/react/caching/cache-interaction/#writequery

Reply:
@Betree : Isn't it even simpler to just update the field itself here since we are actually only concerned about this field in the render method. 🤔 🐨 I've made that change, but let me know if you disagree.

https://github.com/opencollective/opencollective-frontend/blob/5a16808f6fe0cc5a25a0da107c0e2e72efd581e3/components/changelog/ChangelogTrigger.js#L41

Reply:
Apollo's state shouldn't be mutated directly, or there's no guarantee that this state update will be properly dispatched to other components. It's also very likely that another mutation on `LoggedInUser` will reset this field. In some cases, Apollo even throws out exceptions when you try to mutate its state directly.

Reply:
@Betree : I've used `writeQuery` now. Although one thing I noticed is when the cache update happens (using `writeQuery`) the notification disappears, but if the user refresh the page then the notification will appear once. If they refresh again of course the notification will disapper. My understanding is that this is due to https://github.com/opencollective/opencollective/issues/4003 since it needs one more refresh to sync the `LoggedInUser` cache with localstorage? 🤔

Reply:
@Betree : Actually I think I am wrong in my previous logic, reading more this is default behaviour, the `writeQuery` changes are gone when a page refresh is done, as per the documentation; 

```
Any changes you make to cached data with writeQuery are not pushed to your GraphQL server. If you reload your environment, these changes disappear.
```

Reference: https://www.apollographql.com/docs/react/caching/cache-interaction/#using-graphql-queries

Maybe for now we can live with this? Only happens in the first page refresh, if they do again everything will be in sync it seems. 😄 

But there's another way I found, use `LocalStorage` to as the secondary source of truth. This way we can persist the value in localStorage and take it when the refresh occurs. I've made a commit demonstrating this; 
b392cb2. This would solve the problem completely, but let me know your thoughts. 😄

Reply:
> @Betree : I've used `writeQuery` now. Although one thing I noticed is when the cache update happens (using `writeQuery`) the notification disappears, but if the user refresh the page then the notification will appear once. If they refresh again of course the notification will disapper. My understanding is that this is due to [opencollective/opencollective#4003](https://github.com/opencollective/opencollective/issues/4003) since it needs one more refresh to sync the `LoggedInUser` cache with localstorage? 

Correct. It's the reason for https://github.com/opencollective/opencollective/issues/4003, this localStorage cache is not done properly and it's causing many issues like that.

> Actually I think I am wrong in my previous logic, reading more this is default behaviour, the writeQuery changes are gone when a page refresh is done, as per the documentation;
>> Any changes you make to cached data with writeQuery are not pushed to your GraphQL server. If you reload your environment, these changes disappear.


Your previous logic was actually correct. What the documentation means here is that everything done by `writeQuery` only affects your local cache, and nothing is pushed to the API (which is what we want).

> But there's another way I found, use LocalStorage to as the secondary source of truth. This way we can persist the value in localStorage and take it when the refresh occurs. I've made a commit demonstrating this; 8e94fe0. This would solve the problem completely, but let me know your thoughts. smile

Let's not dig into such complex solutions, we can live with this issue for now and look into opencollective/opencollective#4003 later.

Reply:
@Betree : Thanks, I was going over the top here trying to make some kinda fix to make the experience seamless for the user. 😄 Anyways, yes I've reverted that change so we are using `useUser` as before. So I will mark this as solved unless there's any further thing to be done here. 😄